src/initramfs/cmd/init/pkg/mount/mount.go

```diff
@@ -166,16 +169,33 @@ func Unmount() error {
 func mountSpecialDevices() (err error) {
 	for _, mountpoint := range instance.special {
 		if err = os.MkdirAll(mountpoint.target, os.ModeDir); err != nil {
-			return fmt.Errorf("create %s: %s", mountpoint.target, err.Error())
+			return fmt.Errorf("error creating mount point directory %s: %v", mountpoint.target, err)
 		}
-		if err = unix.Mount(mountpoint.source, mountpoint.target, mountpoint.fstype, mountpoint.flags, mountpoint.data); err != nil {
-			return fmt.Errorf("mount %s: %s", mountpoint.target, err.Error())
+		if err = MountWithRetry(mountpoint.source, mountpoint.target, mountpoint.fstype, mountpoint.flags, mountpoint.data); err != nil {
+			return fmt.Errorf("error mounting special device %s: %v", mountpoint.target, err)
 		}
 	}
 
 	return nil
 }
 
+func MountWithRetry(source string, target string, fstype string, flags uintptr, data string) (err error) {
+	for i := 0; i < 10; i++ {
```

Comment:
is 1s (10* 100ms) enough time?

Reply:
Good question. I think this is safe to bump up a bit more. 5 seconds sound good? At this point this code should only be used very early in the boot. But maybe it could be configurable and then usable by `blockd`.